### ISSUE TITLE
SiteSettings and GetDefaultPagesSettings load infinitely if culturecode in PortalLocalization if en-us instead of en-US

### ISSUE THREAD

**Issue by ChrisHammond** [dnnsoftware/Dnn.AdminExperience#886]
## Description of bug
Ran into a problem where my Site Settings and Site Behavior tab would load, and then repeatedly load over and over again, creating thousands of requests to ServicesFramework/GetPortalSettings and GetDefaultPagesSettings, if you left the browser open and persona bar on one of those tabs it will create thousands upon thousands of requests. In my case I have an install with 24 portal, 6 of them had this issue, the others did not. Issue is caused by en-us instead of en-US in the portallocalization table

## Steps to reproduce
List the steps to reproduce the behavior:
1. In PortalLocalization table, update the Culture code from en-US to en-us (why my portals are this way, unsure, but likely due to the site having been around for 10+ years)
2. Restart the app/clear the cache
3. Navigate to the Site Settings in persona bar for the portal in question
4. Open the network tab in dev tools and look at the continual requests.

## Current result
The site continually hits the web services to load site settings over and over again

## Expected result
Site settings should load only one time.

To prevent this, you can update the PortalLocalization table, but that requires updating database data that most people won't be familiar with.

I would think that something inside the platform should adjust for this instead.

Only have this issue on 9.3.1 at this time, but I'll keep an eye out on other sites/installs to see if that's the case in earlier versions.

**Comment by ChrisHammond** [comment]
Quick fix for the bad data if you have this issue

update dbo.PortalLocalization

SET CultureCode = 'en-US'

WHERE CultureCode LIKE 'en-us'

**Comment by mitchelsellers** [comment]
@ChrisHammond Do you know what version this was upgraded from?

**Comment by ChrisHammond** [comment]
Probably 922 to 931. But it started back in the dnn4 days

**Comment by hismightiness** [comment]
Wow... this must've been a beast to track down! Great stuff, Chris! :)

**Comment by ChrisHammond** [comment]
2 days off effort trying to find the code, then I gave up and looked for bad data. Now that I know where the data was off I'll try to find a real fix for the code. PB isn't easy for sure...

**Comment by valadas** [comment]
Nice find. @ChrisHammond if you need any help with the PB, just let me know. The fix here is probably just adding a .tolower somewhere. I will probably get to this for next release but if you want to learn to fish, then just let me know and I will be glad.

**Comment by ChrisHammond** [comment]
@valadas I spent a while last night trying to figure out where to make this change, but I have no idea how any of the bundling stuff works, so I couldn't track it down :(

**Comment by valadas** [comment]
@ChrisHammond I feel you, I had a hard time when I started with the PersonaBar, things have changed a lot and are still changing. I will get to this issue probably for next release, just wanted to mention that if you want a walkthrough on this stuff, just let me know and we can screenshare or something. I owe most of what I know about Dnn to you, so I would return the favor in a flash.

**Comment by efficiondave** [comment]
I ran into this today. Evil things were happening to my entire laptop whenever I opened Site Settings. Eventually found thousands of calls to GetPortalSettings within a matter of seconds. Happened both in 9.2.2 and 9.3.2. This is a pretty big deal in my opinion.

**Comment by valadas** [comment]
Yes, until this is fixed, the workaround is pretty easy to do. That being said, this probably needs to be 2 PRs, one for the UI to handle that special case and another one to fix the underlying issue in the database upon upgrade.
